// Code generated by protoc-gen-validate. DO NOT EDIT.
// source: match.proto

package gen

import (
	"bytes"
	"errors"
	"fmt"
	"net"
	"net/mail"
	"net/url"
	"regexp"
	"sort"
	"strings"
	"time"
	"unicode/utf8"

	"google.golang.org/protobuf/types/known/anypb"
)

// ensure the imports are used
var (
	_ = bytes.MinRead
	_ = errors.New("")
	_ = fmt.Print
	_ = utf8.UTFMax
	_ = (*regexp.Regexp)(nil)
	_ = (*strings.Reader)(nil)
	_ = net.IPv4len
	_ = time.Duration(0)
	_ = (*url.URL)(nil)
	_ = (*mail.Address)(nil)
	_ = anypb.Any{}
	_ = sort.Sort
)

// Validate checks the field values on MatchCalendar with the rules defined in
// the proto definition for this message. If any rules are violated, the first
// error encountered is returned, or nil if there are no violations.
func (m *MatchCalendar) Validate() error {
	return m.validate(false)
}

// ValidateAll checks the field values on MatchCalendar with the rules defined
// in the proto definition for this message. If any rules are violated, the
// result is a list of violation errors wrapped in MatchCalendarMultiError, or
// nil if none found.
func (m *MatchCalendar) ValidateAll() error {
	return m.validate(true)
}

func (m *MatchCalendar) validate(all bool) error {
	if m == nil {
		return nil
	}

	var errors []error

	// no validation rules for Id

	// no validation rules for ClubOneName

	// no validation rules for ClubTwoName

	// no validation rules for MatchDate

	// no validation rules for MatchRound

	if len(errors) > 0 {
		return MatchCalendarMultiError(errors)
	}

	return nil
}

// MatchCalendarMultiError is an error wrapping multiple validation errors
// returned by MatchCalendar.ValidateAll() if the designated constraints
// aren't met.
type MatchCalendarMultiError []error

// Error returns a concatenation of all the error messages it wraps.
func (m MatchCalendarMultiError) Error() string {
	var msgs []string
	for _, err := range m {
		msgs = append(msgs, err.Error())
	}
	return strings.Join(msgs, "; ")
}

// AllErrors returns a list of validation violation errors.
func (m MatchCalendarMultiError) AllErrors() []error { return m }

// MatchCalendarValidationError is the validation error returned by
// MatchCalendar.Validate if the designated constraints aren't met.
type MatchCalendarValidationError struct {
	field  string
	reason string
	cause  error
	key    bool
}

// Field function returns field value.
func (e MatchCalendarValidationError) Field() string { return e.field }

// Reason function returns reason value.
func (e MatchCalendarValidationError) Reason() string { return e.reason }

// Cause function returns cause value.
func (e MatchCalendarValidationError) Cause() error { return e.cause }

// Key function returns key value.
func (e MatchCalendarValidationError) Key() bool { return e.key }

// ErrorName returns error name.
func (e MatchCalendarValidationError) ErrorName() string { return "MatchCalendarValidationError" }

// Error satisfies the builtin error interface
func (e MatchCalendarValidationError) Error() string {
	cause := ""
	if e.cause != nil {
		cause = fmt.Sprintf(" | caused by: %v", e.cause)
	}

	key := ""
	if e.key {
		key = "key for "
	}

	return fmt.Sprintf(
		"invalid %sMatchCalendar.%s: %s%s",
		key,
		e.field,
		e.reason,
		cause)
}

var _ error = MatchCalendarValidationError{}

var _ interface {
	Field() string
	Reason() string
	Key() bool
	Cause() error
	ErrorName() string
} = MatchCalendarValidationError{}

// Validate checks the field values on CreateMatchCalendarResponse with the
// rules defined in the proto definition for this message. If any rules are
// violated, the first error encountered is returned, or nil if there are no violations.
func (m *CreateMatchCalendarResponse) Validate() error {
	return m.validate(false)
}

// ValidateAll checks the field values on CreateMatchCalendarResponse with the
// rules defined in the proto definition for this message. If any rules are
// violated, the result is a list of violation errors wrapped in
// CreateMatchCalendarResponseMultiError, or nil if none found.
func (m *CreateMatchCalendarResponse) ValidateAll() error {
	return m.validate(true)
}

func (m *CreateMatchCalendarResponse) validate(all bool) error {
	if m == nil {
		return nil
	}

	var errors []error

	// no validation rules for Id

	if len(errors) > 0 {
		return CreateMatchCalendarResponseMultiError(errors)
	}

	return nil
}

// CreateMatchCalendarResponseMultiError is an error wrapping multiple
// validation errors returned by CreateMatchCalendarResponse.ValidateAll() if
// the designated constraints aren't met.
type CreateMatchCalendarResponseMultiError []error

// Error returns a concatenation of all the error messages it wraps.
func (m CreateMatchCalendarResponseMultiError) Error() string {
	var msgs []string
	for _, err := range m {
		msgs = append(msgs, err.Error())
	}
	return strings.Join(msgs, "; ")
}

// AllErrors returns a list of validation violation errors.
func (m CreateMatchCalendarResponseMultiError) AllErrors() []error { return m }

// CreateMatchCalendarResponseValidationError is the validation error returned
// by CreateMatchCalendarResponse.Validate if the designated constraints
// aren't met.
type CreateMatchCalendarResponseValidationError struct {
	field  string
	reason string
	cause  error
	key    bool
}

// Field function returns field value.
func (e CreateMatchCalendarResponseValidationError) Field() string { return e.field }

// Reason function returns reason value.
func (e CreateMatchCalendarResponseValidationError) Reason() string { return e.reason }

// Cause function returns cause value.
func (e CreateMatchCalendarResponseValidationError) Cause() error { return e.cause }

// Key function returns key value.
func (e CreateMatchCalendarResponseValidationError) Key() bool { return e.key }

// ErrorName returns error name.
func (e CreateMatchCalendarResponseValidationError) ErrorName() string {
	return "CreateMatchCalendarResponseValidationError"
}

// Error satisfies the builtin error interface
func (e CreateMatchCalendarResponseValidationError) Error() string {
	cause := ""
	if e.cause != nil {
		cause = fmt.Sprintf(" | caused by: %v", e.cause)
	}

	key := ""
	if e.key {
		key = "key for "
	}

	return fmt.Sprintf(
		"invalid %sCreateMatchCalendarResponse.%s: %s%s",
		key,
		e.field,
		e.reason,
		cause)
}

var _ error = CreateMatchCalendarResponseValidationError{}

var _ interface {
	Field() string
	Reason() string
	Key() bool
	Cause() error
	ErrorName() string
} = CreateMatchCalendarResponseValidationError{}

// Validate checks the field values on SuccessMatchResponse with the rules
// defined in the proto definition for this message. If any rules are
// violated, the first error encountered is returned, or nil if there are no violations.
func (m *SuccessMatchResponse) Validate() error {
	return m.validate(false)
}

// ValidateAll checks the field values on SuccessMatchResponse with the rules
// defined in the proto definition for this message. If any rules are
// violated, the result is a list of violation errors wrapped in
// SuccessMatchResponseMultiError, or nil if none found.
func (m *SuccessMatchResponse) ValidateAll() error {
	return m.validate(true)
}

func (m *SuccessMatchResponse) validate(all bool) error {
	if m == nil {
		return nil
	}

	var errors []error

	if all {
		switch v := interface{}(m.GetData()).(type) {
		case interface{ ValidateAll() error }:
			if err := v.ValidateAll(); err != nil {
				errors = append(errors, SuccessMatchResponseValidationError{
					field:  "Data",
					reason: "embedded message failed validation",
					cause:  err,
				})
			}
		case interface{ Validate() error }:
			if err := v.Validate(); err != nil {
				errors = append(errors, SuccessMatchResponseValidationError{
					field:  "Data",
					reason: "embedded message failed validation",
					cause:  err,
				})
			}
		}
	} else if v, ok := interface{}(m.GetData()).(interface{ Validate() error }); ok {
		if err := v.Validate(); err != nil {
			return SuccessMatchResponseValidationError{
				field:  "Data",
				reason: "embedded message failed validation",
				cause:  err,
			}
		}
	}

	// no validation rules for Message

	if len(errors) > 0 {
		return SuccessMatchResponseMultiError(errors)
	}

	return nil
}

// SuccessMatchResponseMultiError is an error wrapping multiple validation
// errors returned by SuccessMatchResponse.ValidateAll() if the designated
// constraints aren't met.
type SuccessMatchResponseMultiError []error

// Error returns a concatenation of all the error messages it wraps.
func (m SuccessMatchResponseMultiError) Error() string {
	var msgs []string
	for _, err := range m {
		msgs = append(msgs, err.Error())
	}
	return strings.Join(msgs, "; ")
}

// AllErrors returns a list of validation violation errors.
func (m SuccessMatchResponseMultiError) AllErrors() []error { return m }

// SuccessMatchResponseValidationError is the validation error returned by
// SuccessMatchResponse.Validate if the designated constraints aren't met.
type SuccessMatchResponseValidationError struct {
	field  string
	reason string
	cause  error
	key    bool
}

// Field function returns field value.
func (e SuccessMatchResponseValidationError) Field() string { return e.field }

// Reason function returns reason value.
func (e SuccessMatchResponseValidationError) Reason() string { return e.reason }

// Cause function returns cause value.
func (e SuccessMatchResponseValidationError) Cause() error { return e.cause }

// Key function returns key value.
func (e SuccessMatchResponseValidationError) Key() bool { return e.key }

// ErrorName returns error name.
func (e SuccessMatchResponseValidationError) ErrorName() string {
	return "SuccessMatchResponseValidationError"
}

// Error satisfies the builtin error interface
func (e SuccessMatchResponseValidationError) Error() string {
	cause := ""
	if e.cause != nil {
		cause = fmt.Sprintf(" | caused by: %v", e.cause)
	}

	key := ""
	if e.key {
		key = "key for "
	}

	return fmt.Sprintf(
		"invalid %sSuccessMatchResponse.%s: %s%s",
		key,
		e.field,
		e.reason,
		cause)
}

var _ error = SuccessMatchResponseValidationError{}

var _ interface {
	Field() string
	Reason() string
	Key() bool
	Cause() error
	ErrorName() string
} = SuccessMatchResponseValidationError{}

// Validate checks the field values on SuccessMatchResponse_Data with the rules
// defined in the proto definition for this message. If any rules are
// violated, the first error encountered is returned, or nil if there are no violations.
func (m *SuccessMatchResponse_Data) Validate() error {
	return m.validate(false)
}

// ValidateAll checks the field values on SuccessMatchResponse_Data with the
// rules defined in the proto definition for this message. If any rules are
// violated, the result is a list of violation errors wrapped in
// SuccessMatchResponse_DataMultiError, or nil if none found.
func (m *SuccessMatchResponse_Data) ValidateAll() error {
	return m.validate(true)
}

func (m *SuccessMatchResponse_Data) validate(all bool) error {
	if m == nil {
		return nil
	}

	var errors []error

	// no validation rules for ClubOneName

	// no validation rules for ClubTwoName

	// no validation rules for MatchDate

	// no validation rules for MatchRound

	if len(errors) > 0 {
		return SuccessMatchResponse_DataMultiError(errors)
	}

	return nil
}

// SuccessMatchResponse_DataMultiError is an error wrapping multiple validation
// errors returned by SuccessMatchResponse_Data.ValidateAll() if the
// designated constraints aren't met.
type SuccessMatchResponse_DataMultiError []error

// Error returns a concatenation of all the error messages it wraps.
func (m SuccessMatchResponse_DataMultiError) Error() string {
	var msgs []string
	for _, err := range m {
		msgs = append(msgs, err.Error())
	}
	return strings.Join(msgs, "; ")
}

// AllErrors returns a list of validation violation errors.
func (m SuccessMatchResponse_DataMultiError) AllErrors() []error { return m }

// SuccessMatchResponse_DataValidationError is the validation error returned by
// SuccessMatchResponse_Data.Validate if the designated constraints aren't met.
type SuccessMatchResponse_DataValidationError struct {
	field  string
	reason string
	cause  error
	key    bool
}

// Field function returns field value.
func (e SuccessMatchResponse_DataValidationError) Field() string { return e.field }

// Reason function returns reason value.
func (e SuccessMatchResponse_DataValidationError) Reason() string { return e.reason }

// Cause function returns cause value.
func (e SuccessMatchResponse_DataValidationError) Cause() error { return e.cause }

// Key function returns key value.
func (e SuccessMatchResponse_DataValidationError) Key() bool { return e.key }

// ErrorName returns error name.
func (e SuccessMatchResponse_DataValidationError) ErrorName() string {
	return "SuccessMatchResponse_DataValidationError"
}

// Error satisfies the builtin error interface
func (e SuccessMatchResponse_DataValidationError) Error() string {
	cause := ""
	if e.cause != nil {
		cause = fmt.Sprintf(" | caused by: %v", e.cause)
	}

	key := ""
	if e.key {
		key = "key for "
	}

	return fmt.Sprintf(
		"invalid %sSuccessMatchResponse_Data.%s: %s%s",
		key,
		e.field,
		e.reason,
		cause)
}

var _ error = SuccessMatchResponse_DataValidationError{}

var _ interface {
	Field() string
	Reason() string
	Key() bool
	Cause() error
	ErrorName() string
} = SuccessMatchResponse_DataValidationError{}
